### PR TITLE
Prometheus grafana

### PR DESCRIPTION
**Pull request by Adam-D-Lewis**
Initial integration of prometheus/grafana into qhub.  I tested it locally and it seems to be working. The kubernetes tests appear to be broken for an unrelated reason.

### CODE REVIEW

File: qhub/template/{{ cookiecutter.repo_directory }}/infrastructure/modules/kubernetes/ingress/main.tf [hunk 1]
```diff
@@ -191,7 +191,7 @@ resource "kubernetes_deployment" "main" {
             "--entrypoints.web.http.redirections.entryPoint.scheme=https",
             # Enable debug logging. Useful to work out why something might not be
             # working. Fetch logs of the pod.
-            "--log.level=${var.loglevel}",
+            "--log.level=DEBUG", #${var.loglevel}",
```
Comment (Adam-D-Lewis) [inline qhub/template/{{ cookiecutter.repo_directory }}/infrastructure/modules/kubernetes/ingress/main.tf:194]: oops, didn't mean to commit this.  I'll undo this.

**Comment by costrouc** [comment]
I'd prefer if we could pull in the helm charts similar to https://github.com/Quansight/qhub/blob/main/qhub/template/%7B%7B%20cookiecutter.repo_directory%20%7D%7D/infrastructure/modules/kubernetes/services/jupyterhub/main.tf#L6. This still pins the chart to a particular version. Is there a reason that we are copying the entire chart locally?

**Comment by Adam-D-Lewis** [comment]
@costrouc I was following the example of what was done for the clearml (https://github.com/Quansight/qhub/tree/main/qhub/template/%7B%7B%20cookiecutter.repo_directory%20%7D%7D/infrastructure/modules/kubernetes/services/clearml), but I don't think there is anything preventing us from importing the chart.  @aktech any thoughts for or against putting the entire helm chart in the repo?

**Comment by aktech** [comment]
@Adam-D-Lewis I agree with @costrouc we shouldn't put helm charts in the repository. ClearML was a special case, as we had to modify the chart. Also, since the last update on ClearML, I'll see if we can remove the ClearML charts as well.

**Comment by Adam-D-Lewis** [comment]
@costrouc ready for a re-review when you are able, thanks!

**Comment by aktech** [comment]
Do you know why is the Kubernetes tests failing?

File: qhub/template/{{ cookiecutter.repo_directory }}/infrastructure/modules/kubernetes/services/monitoring/values-monitoring-qhub.yaml [hunk 2]
```diff
@@ -0,0 +1,2532 @@
...
+  ## Deploy default dashboards.
+  ##
+  defaultDashboardsEnabled: true
+
+  adminPassword: prom-operator
+  
+  grafana.ini:
+    server:
+     # The full public facing url you use in browser, used for redirects and emails
+     domain: github-actions.qhub.dev
```
Comment (aktech) [inline qhub/template/{{ cookiecutter.repo_directory }}/infrastructure/modules/kubernetes/services/monitoring/values-monitoring-qhub.yaml:633]: I believe this has to be custom for each deployment. This is

**Review by aktech — CHANGES_REQUESTED** [review]
It seems like the following files can be removed:

- `values-monitoring-qhub.yaml`
- `values-original.yaml`

Also the diff between the original and overridden files is just following:

```diff
❯ diff values-original.yaml values-monitoring-qhub.yaml
628a629,637
>
>   grafana.ini:
>     server:
>      # The full public facing url you use in browser, used for redirects and emails
>      domain: github-actions.qhub.dev
>      root_url: "%(protocol)s://%(domain)s/monitoring"
>      server_from_sub_path: true
>      router_logging: true
>      enable_gzip: true
732c741
<
---
>
```

Which means you can set the particular values via [terraform's helm `set` attribute](https://registry.terraform.io/providers/hashicorp/helm/latest/docs#example-usage).

File: docs/source/introduction.md [hunk 3]
```diff
@@ -43,6 +43,8 @@ with.
 
 + [**prefect**](https://www.prefect.io/) workflow management
 + [**clearml**](https://clear.ml/) machine learning platform
++ [**prometheus**](https://grafana.com/) cluster monitoring
```
Comment (aktech) [inline docs/source/introduction.md:46]: typo in the url for prometheus

Comment (Adam-D-Lewis) [reply docs/source/introduction.md]: fixed

**Comment by Adam-D-Lewis** [comment]
>It seems like the following files can be removed:

>values-monitoring-qhub.yaml
values-original.yaml
Also the diff between the original and overridden files is just following:

❯ diff values-original.yaml values-monitoring-qhub.yaml
628a629,637
>
>   grafana.ini:
>     server:
>      # The full public facing url you use in browser, used for redirects and emails
>      domain: github-actions.qhub.dev
>      root_url: "%(protocol)s://%(domain)s/monitoring"
>      server_from_sub_path: true
>      router_logging: true
>      enable_gzip: true
732c741
<
---
>
>Which means you can set the particular values via terraform's helm set attribute.

I made the requested changes.

**Comment by aktech** [comment]
Waiting for the tests to pass here: https://github.com/Quansight/qhub/pull/748 to see if the failed K8s tests in here are relevant or not.

**Comment by aktech** [comment]
Seems like the tests pass on main, can you check why k8s tests fail? I can rerun again to confirm,

**Comment by Adam-D-Lewis** [comment]
Cyprus says there is insufficient cpu on the node with the additional pods in this PR.  Maybe I can reduce the used cpu or maybe we can use CIrun

**Comment by Adam-D-Lewis** [comment]
Blocked by https://github.com/Quansight/qhub/issues/756

**Comment by costrouc** [comment]
Could you add an entry to the changelog?

**Review by costrouc — APPROVED** [review]

**Pull request merged**